netbox/netbox/authentication.py

```diff
@@ -323,6 +324,52 @@ def get_permission_filter(self, user_obj):
                     hasattr(user_obj.ldap_user, "group_names")):
                 permission_filter = permission_filter | Q(groups__name__in=user_obj.ldap_user.group_names)
             return permission_filter
+
+    # Monkey-patch _mirror_groups, code is from django-auth-ldap.backends._LDAPUser
+    # There are no changes to this routine, the 'fix' is the import of Group above.
+    def _mirror_groups(self):
+        """
+        Mirrors the user's LDAP groups in the Django database and updates the
+        user's membership.
+        """
+        target_group_names = frozenset(self._get_groups().get_group_names())
+        target_group_names = frozenset("testgroup",)
+        current_group_names = frozenset(
+            self._user.groups.values_list("name", flat=True).iterator()
+        )
+
+        # These were normalized to sets above.
+        MIRROR_GROUPS_EXCEPT = self.settings.MIRROR_GROUPS_EXCEPT
+        MIRROR_GROUPS = self.settings.MIRROR_GROUPS
+
+        # If the settings are white- or black-listing groups, we'll update
+        # target_group_names such that we won't modify the membership of groups
+        # beyond our purview.
+        if isinstance(MIRROR_GROUPS_EXCEPT, (set, frozenset)):
+            target_group_names = (target_group_names - MIRROR_GROUPS_EXCEPT) | (
+                current_group_names & MIRROR_GROUPS_EXCEPT
+            )
+        elif isinstance(MIRROR_GROUPS, (set, frozenset)):
+            target_group_names = (target_group_names & MIRROR_GROUPS) | (
+                current_group_names - MIRROR_GROUPS
+            )
+
+        if target_group_names != current_group_names:
+            existing_groups = list(
+                Group.objects.filter(name__in=target_group_names).iterator()
+            )
+            existing_group_names = frozenset(group.name for group in existing_groups)
+
+            new_groups = [
+                Group.objects.get_or_create(name=name)[0]
+                for name in target_group_names
+                if name not in existing_group_names
+            ]
+
+            self._user.groups.set(existing_groups + new_groups)
```

Comment:
I'd guess that this would be also OK (performance wise) if you don't do a single query for the existing groups, but simplify this to
```suggestion
            target_groups = [
                Group.objects.get_or_create(name=name)[0]
                for name in target_group_names
            ]

            self._user.groups.set(target_groups)
```

Reply:
Yes, it's just exactly the same code as in django-auth-ldap, I didn't want to change it so it is easy to merge in any changes in django-auth-ldap in the future.  The only difference is the import statement above it to switch to the NetBox Group instead of the Django.auth one.